extensions/cornerstone-dicom-rt/src/loadRTStruct.js

```diff
@@ -55,42 +56,54 @@ async function checkAndLoadContourData(instance, datasource) {
 
   // Modify contourData and replace it in its corresponding ROIContourSequence's Contour's contourData
   let index = 0;
-  instance.ROIContourSequence.forEach((ROIContour, roiIndex) => {
-    ROIContour.ContourSequence.forEach((Contour, contourIndex) => {
-      const promise = resolvedPromises[index++];
-
-      if (promise.status === 'fulfilled') {
-        const uint8Array = new Uint8Array(promise.value);
-        const textDecoder = new TextDecoder();
-        const dataUint8Array = textDecoder.decode(uint8Array);
-        if (
-          typeof dataUint8Array === 'string' &&
-          dataUint8Array.includes('\\')
-        ) {
-          const numSlashes = (dataUint8Array.match(/\\/g) || []).length;
-          let startIndex = 0;
-          let endIndex = dataUint8Array.indexOf('\\', startIndex);
-          let numbersParsed = 0;
-          const ContourData = [];
-
-          while (numbersParsed !== numSlashes + 1) {
-            const str = dataUint8Array.substring(startIndex, endIndex);
-            let value = parseFloat(str);
-
-            ContourData.push(value);
-            startIndex = endIndex + 1;
-            endIndex = dataUint8Array.indexOf('\\', startIndex);
-            endIndex === -1 ? (endIndex = dataUint8Array.length) : endIndex;
-            numbersParsed++;
+  instance.ROIContourSequence.forEach(ROIContour => {
+    try {
+      ROIContour.ContourSequence.forEach((Contour, contourIndex) => {
+        const promise = resolvedPromises[index++];
```

Comment:
Could this be changed to a map by contourIndex?  I'm just concerned that we might get out of sync on this and end up with the wrong contour somewhere.  Mapping the promises definitely rather than by position just feels safer.